incompatibility/incomp-scanning.go

```diff
@@ -80,10 +120,32 @@ func ScanIncompatibilities(document *openapiv3.Document, reportIdentifier string
 	return ReportOnDoc(document, reportIdentifier, IncompatibilityReporters...)
 }
 
+func newIDReport(reportIdentifier string, incompDescriptions []*IncompatibilityDescription) *IDReport {
+	return &IDReport{
+		ReportIdentifier:  reportIdentifier,
+		Incompatibilities: incompDescriptions,
+	}
+}
+
 func newIncompatibility(severity Severity, classification IncompatibiltiyClassification, path ...string) *Incompatibility {
```

Comment:
Is this used?

Reply:
not sure which piece you're refering to here but the incompatibilities field is how we attach the detailed incompatibilities to a report and newIncompatibility used whenever the base incompatibility creates a new incompatibility it wishes to attach to a report

Reply:
I see this is a private function, but I don't see it used in this package.

Reply:
moved to reporter.go